subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishDependenciesIntegTest.groovy

```diff
@@ -158,11 +158,15 @@ class MavenPublishDependenciesIntegTest extends AbstractIntegrationSpec {
 
         where:
         plugin         | gradleConfiguration  | mavenScope
-        'java'         | 'compile'            | 'runtime'
+        'java'         | 'compile'            | 'compile'
+        'java'         | 'runtime'            | 'compile'
```

Comment:
That's exactly why 'compile' and 'runtime' are deprecated, they have confusing behavior no matter what we do. If you want non-confusing behavior, you need to use 'api', 'implementation' and 'runtimeOnly' instead.

All we're doing here is making the behavior consistent between 'java' and 'java-library' and local dependencies and external dependencies. So at least a users sees the same behavior no matter which plugin he's using and no matter if the consumer is a project in the same build or if it is a project in a different repository consuming through Maven metadata.